src/js/base/handalone.js

```diff
@@ -147,34 +150,45 @@ require(["pyret-base/js/runtime", "program"], function(runtimeLib, program) {
       runtime.setParam("current-checker", currentChecker);
     }
   };
-  postLoadHooks[main] = function(answer) {
-    var checkerLib = runtime.modules["builtin://checker"];
-    var checker = runtime.getField(runtime.getField(checkerLib, "provide-plus-types"), "values");
-    var getStack = function(err) {
-      console.error("The error is: ", err);
-      var locArray = err.val.pyretStack.map(runtime.makeSrcloc);
-      var locList = runtime.ffi.makeList(locArray);
-      return locList;
+  if (runMode) {
+    postLoadHooks[main] = function(answer) {
+      var definedValues = runtime.getField(answer, "defined-values");
+      var definedExitCode = definedValues["exit-code"];
```

Comment:
So this is a new implicit interface that if a module provides an exit-code field, and it's the main module of a run, that value is used as the exit code for the whole run?

I don't know that this is bad, but it seems somewhat unique.

How do other languages do this?  Can we borrow a different design?  What makes us think this is the right design?